types/react/index.d.ts

```diff
@@ -282,7 +282,7 @@ declare namespace React {
         // tslint:enable:unified-signatures
 
         forceUpdate(callBack?: () => any): void;
-        render(): JSX.Element | null | false;
+        render(): JSX.Element | JSX.Element[] | string | number | null | false;
```

Comment:
Maybe `React.ReactNode` is more accurate (though I'm not sure how that will interact with JSX). `JSX.Element[]` doesn't capture the fact that the array is allowed to contain non-element values.

Reply:
Maybe adding `React.ReactNode` would fix the following pattern. I'm currently getting an error in TS right now but I believe React allows us to have one item that is `null` to remove the node.

```jsx
render() {
  return [
    <button key="modal-button" onClick={this.openModal}>
        Open modal
    </button>,
    this.state.shouldRenderModal ? (
      <Modal />
    ) : null,
  ];
}
```

As a temporary fix I can do

```diff
render() {
  return [
    <button key="modal-button" onClick={this.openModal}>
        Open modal
      </button>,
-      this.state.shouldRenderModal ? (
+      (this.state.shouldRenderModal ? (
       <Modal />
-      ) : null,
+      ) : null) as JSX.Element,
  ];
}
```